### PR TITLE
Did peer refactoring, new display macro

### PR DESCRIPTION
**Pull request by Patrik-Stas**
# Display
- Create new macro `Display` for deriving `Display` which display structure as json. Located in shared reusable crate `tools/display_as_json`
- Derive `Display` implementation `DidDoc` and its underlying components

# APIs: Building did_peers
The changes overall are trying to make the API more concise and grouped together, letting user operate directly on PeerDid<N> type, rather than calling isolated util functions. Additionally making sure that user won't ever operate on Numalgo type(s) itself, but work with user facing `PeerDid<N>` only.

#### Building `PeerDid::<Numalgo3>` from `DidDoc`
Removed util function
```
generate_numalgo3(did_doc)
```
in favor of new API:
```
PeerDid::<Numalgo3>::from_did_document(did_doc)
```

#### Building `PeerDid::<Numalgo2>` from `DidDoc`
Equivalent of above for numalgo2; removed function `generate_numalgo2(did_doc)`

#### Building `PeerDid::<Numalgo3>` from `Did`
Removed util function
```
generate_numalgo3(did)
```
as existing API
```
PeerDid::<Numalgo3>::parse(did)
```
already serves the same purpose

#### Building `PeerDid<Numalgo3>` from `PeerDid<Numalgo2>` 
Removed `ToNumalgo3` trait in favor of specializing `PeerDid<Numalgo2>` with function `to_numalgo3(&self)`. The API is unchanged, eg:
```
peer_did_2.to_numalgo3()
```

# `did_peer` crate onboarding
- Added `readme.md` with some information about the crate
- Added example `did_peer/examples/demo.rs`

### CODE REVIEW

File: did_doc_sov/src/lib.rs [hunk 1]
```diff
@@ -1,3 +1,5 @@
+extern crate display_as_json;
```
Comment (mirgee) [inline did_doc_sov/src/lib.rs:1]: This should not be necessary

File: did_peer/src/numalgos/numalgo2/mod.rs [hunk 2]
```diff
@@ -1,8 +1,42 @@
...
+impl FromDidDoc for Numalgo2 {
+    fn from_did_doc(
+        did_document: DidDocument<ExtraFieldsSov>,
+    ) -> Result<PeerDid<Numalgo2>, DidPeerError> {
+        let mut did = String::from("did:peer:2");
+
+        did = append_encoded_key_segments(did, &did_document)?;
+        println!("after append_encoded_key_segments >> did: {}", did);
+        did = append_encoded_service_segment(did, &did_document)?;
+        println!("after append_encoded_service_segment >> did: {}", did);
```
Comment (mirgee) [inline did_peer/src/numalgos/numalgo2/mod.rs:30]: Should be debug / trace level or removed

File: did_peer/src/peer_did/numalgos/numalgo2.rs [hunk 3]
```diff
@@ -29,9 +24,3 @@ impl ResolvableNumalgo for Numalgo2 {
         resolve_numalgo2(did, public_key_encoding).map(|builder| builder.build())
     }
 }
-
-impl ToNumalgo3 for Numalgo2 {
-    fn to_numalgo3(did: &Did) -> Result<PeerDid<Numalgo3>, DidPeerError> {
-        generate_numalgo3(did)
-    }
-}
```
Comment (mirgee) [inline did_peer/src/peer_did/numalgos/numalgo2.rs:37]: Why was this removed?

File: did_peer/tests/demo.rs [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+};
+use did_doc_sov::extra_fields::{didcommv1::ExtraFieldsDidCommV1, ExtraFieldsSov, KeyKind};
+use did_parser::{Did, DidUrl};
+use did_peer::peer_did::{
+    numalgos::{numalgo2::Numalgo2, numalgo3::Numalgo3},
+    PeerDid,
+};
+
+#[test]
+fn demo() -> Result<(), Box<dyn Error>> {
```
Comment (mirgee) [inline did_peer/tests/demo.rs:17]: I would prefer avoiding test duplication by removing this (and potentially expanding the test macros, if you think it's worth it).

Comment (Patrik-Stas) [reply did_peer/tests/demo.rs]: I've moved this to `examples` directory which is quite common practice, added readme etc. Given it's now incorporated into readme as part of crate onboarding, I think you will find its existence more justified.

File: did_peer/README.md
Comment (mirgee) [inline did_peer/README.md:1]: I think the readme should definitely include link to the spec: https://identity.foundation/peer-did-method-spec

**Review by mirgee — APPROVED** [review]
